test/test_tensordict.py

```diff
@@ -5906,6 +5907,46 @@ def test_empty():
     assert len(list(td_empty.get("b").keys())) == 1
 
 
+@pytest.mark.parametrize(
+    "stack_dim",
+    [0, 1, 2, 3],
+)
+@pytest.mark.parametrize(
+    "nested_stack_dim",
+    [0, 1, 2],
+)
+def test_dense_stack_tds(stack_dim, nested_stack_dim):
```

Comment:
I don't see what we're testing specific to this method here. All the assertions would hold with a regular tensordict no?
Why aren't we testing anything specific to the keys we provide?

Reply:
the idea is that this function should behave exactly the same with regualr tds or lazy stacks so the fact that a regualr td passes is ok to me. the main things we are testing is that when stacking tds that contain lazy stacks we can call this function to remove one level of lazyness and densify one stack_dim